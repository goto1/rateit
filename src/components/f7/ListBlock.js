import React from "react";
import PropTypes from "prop-types";

const ListBlock = ({ children }) =>
  <div className="list-block">
    <ul>
      {children}
    </ul>
  </div>;

ListBlock.propTypes = {
  children: PropTypes.object.isRequired
};

export default ListBlock;
