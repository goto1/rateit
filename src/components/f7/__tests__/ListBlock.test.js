import React from "react";
import ListBlock from "../ListBlock";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import "jest-enzyme";

describe("ListBlock component", () => {
  const HelloWorld = <div>HelloWorld!</div>;

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ListBlock>
          {HelloWorld}
        </ListBlock>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("contains a component", () => {
    const wrapper = shallow(
      <ListBlock>
        {HelloWorld}
      </ListBlock>
    );
    expect(wrapper).toContainReact(HelloWorld);
  });
});
