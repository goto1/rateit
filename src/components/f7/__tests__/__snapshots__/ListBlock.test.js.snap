// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBlock component renders correctly 1`] = `


<div
  className="list-block"
>
  <ul>
    <div>
      HelloWorld!
    </div>
  </ul>
</div>
`;
